int32 piece
string cell
---
